index.d.ts

```diff
@@ -18,7 +18,7 @@ declare namespace unless {
          * and methods key-pairs. If the request path or path and method match, the middleware will not run. Check
          * Examples for usage.
          */
-        path?: string | string[];
+        path?: string | RegExp | (string | RegExp | { url: (string | RegExp); methods: string[]; })[];
```

Comment:
In that case I would say we should define a type alias so we can reference it recursively.